### PR TITLE
Rm2021/mobile/atchyuth/re tasks

### PR DESCRIPTION
**Pull request by Atchyutkumar-cmd**

### CODE REVIEW

File: Atchyuth/adb_install.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    res=None
+    try:
+        res = subprocess.check_output(cmd)
+        res = res.decode('utf-8')
+    except Exception as e:
+        print("Error : {}".format(str(e)))
+    return res
+cmd='adb install C:\Andriod\platform-tools\cal.apk'
+cmd='adb install C:\Andriod\platform-tools\python_sample_mobile_app.apk'
+res=run_adb(cmd)
```
Comment (mupputur) [inline Atchyuth/adb_install.py:13]: As discussed the test code should be deleted or commented or separated when we are implmneting a modules.
For example
if __name__ == "__main__":
   cmd='adb install C:\Andriod\platform-tools\cal.apk'
   cmd='adb install C:\Andriod\platform-tools\python_sample_mobile_app.apk'
   res=run_adb(cmd)
   res=run_adb(cmd)
   print(res)
   print(res)

File: Atchyuth/adb_install.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import subprocess
+import os
+def run_adb(cmd_list):
```
Comment (mupputur) [inline Atchyuth/adb_install.py:3]: function names shoud be meaningful

In this case you can use install_app()

File: Atchyuth/adb_install.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    res=None
```
Comment (mupputur) [inline Atchyuth/adb_install.py:4]: As function is meant for installation , you can hardcode the command inside
def install_app(apk_path):
    res=None
    cmd = "adb install " + apk_path

File: Atchyuth/adb_install.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    try:
+        res = subprocess.check_output(cmd)
+        res = res.decode('utf-8')
+    except Exception as e:
+        print("Error : {}".format(str(e)))
+    return res
+cmd='adb install C:\Andriod\platform-tools\cal.apk'
```
Comment (mupputur) [inline Atchyuth/adb_install.py:11]: Why you are installing same app twice ?

File: Atchyuth/adb_install.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import subprocess
+import os
+def run_adb(cmd_list):
+    res=None
+    try:
+        res = subprocess.check_output(cmd)
+        res = res.decode('utf-8')
+    except Exception as e:
```
Comment (mupputur) [inline Atchyuth/adb_install.py:8]: Wrong exception handling

File: Atchyuth/adb_install.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import subprocess
+import os
+def run_adb(cmd_list):
+    res=None
+    try:
+        res = subprocess.check_output(cmd)
```
Comment (mupputur) [inline Atchyuth/adb_install.py:6]: This code can cause multiple exception. Please handle the excpected exceptions explicitly.

**Review by mupputur — CHANGES_REQUESTED** [review]

File: Atchyuth/adb_install.py (hunk 1 above)
Comment (Atchyutkumar-cmd) [reply Atchyuth/adb_install.py]: code is  modified as per suggestion

File: Atchyuth/adb_install.py (hunk 3 above)
Comment (Atchyutkumar-cmd) [reply Atchyuth/adb_install.py]: modified the code

File: Atchyuth/adb_install.py (hunk 6 above)
Comment (Atchyutkumar-cmd) [reply Atchyuth/adb_install.py]: Handled the expected exceptions explicitly in the code

File: Atchyuth/adb_install.py (hunk 5 above)
Comment (Atchyutkumar-cmd) [reply Atchyuth/adb_install.py]: Modified the code

File: Atchyuth/1_install_apk.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import os
+import subprocess
+def install_apk(apk_path):
+    try:
+        apk=apk_path.endswith(".apk")
+        if apk==True:
```
Comment (mupputur) [inline Atchyuth/1_install_apk.py:6]: if not apk_path.endswith(".apk"):
   raise Exception("The file format should ends with .apk")

File: Atchyuth/2_t2_ex1_current_running_processes.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+def cur_run_processes():
+    pro_str=''
+    for process in psutil.process_iter():
+        pro_name=process.name()
+        Id=process.pid
+        process=str(Id)+' '+pro_name
+        pro_str=pro_str+'  '+process
+    fobj=open(path,'w')
+    pro_file=fobj.write(pro_str)
+    fobj.close()
```
Comment (mupputur) [inline Atchyuth/2_t2_ex1_current_running_processes.py:13]: Please remove this and create a separate PR

**Review by mupputur — APPROVED** [review]
Approved

**Pull request merged**